### PR TITLE
Add Topic description into api.yaml

### PR DESCRIPTION
**Pull request by Ielay**
Issue [#308](https://github.com/malinink/easy-tests/issues/308)

### CODE REVIEW

**Comment by Ielay** [comment]
@malinink 
Ready to review

File: src/main/resources/api/api.yaml [hunk 1]
```diff
@@ -186,6 +186,132 @@ paths:
...
+  /v1/topics:
+    'get':
+      summary: List all topics
+      operationId: listTopics
+      tags:
+        - topics
+      responses:
+        '200':
+          description: An array of topics
+          schema:
```
Comment (malinink) [inline src/main/resources/api/api.yaml]: use `V1Topics`

File: src/main/resources/api/api.yaml [hunk 2]
```diff
@@ -186,6 +186,132 @@ paths:
...
+            type: array
+            items:
+              $ref: '#/definitions/V1Topic'
+        '403':
```
Comment (malinink) [inline src/main/resources/api/api.yaml]: add 400, 404 errors codes (look users)

File: src/main/resources/api/api.yaml [hunk 3]
```diff
@@ -186,6 +186,132 @@ paths:
   #
   # Topics
   #
+  /v1/topics:
```
Comment (malinink) [inline src/main/resources/api/api.yaml]: where is query parameter?

File: src/main/resources/api/api.yaml [hunk 4]
```diff
@@ -186,6 +186,132 @@ paths:
...
+          required: true
+          description: The topic that is created
+          schema:
+            $ref: '#/definitions/V1Topic'
+      tags:
+        - topics
+      responses:
+        '201':
+          description: The topic was successfully created
+        '400':
```
Comment (malinink) [inline src/main/resources/api/api.yaml]: order responses codes

File: src/main/resources/api/api.yaml [hunk 5]
```diff
@@ -186,6 +186,132 @@ paths:
...
+          description: Bad request
+          schema:
+            $ref: '#/definitions/V1Error'
+        '401':
+          description: User isn't authenticated
```
Comment (malinink) [inline src/main/resources/api/api.yaml:245]: ref?

File: src/main/resources/api/api.yaml [hunk 6]
```diff
@@ -186,6 +186,132 @@ paths:
...
+          in: body
+          required: true
+          description: The topic that is created
+          schema:
+            $ref: '#/definitions/V1Topic'
+      tags:
+        - topics
+      responses:
+        '201':
+          description: The topic was successfully created
```
Comment (malinink) [inline src/main/resources/api/api.yaml]: 201 should return identity model

File: src/main/resources/api/api.yaml [hunk 7]
```diff
@@ -186,6 +186,132 @@ paths:
...
+        - in: body
+          name: topic
+          description: The topic that needs to be added instead of the existing one with the same id
+          required: true
+          schema:
+            $ref: '#/definitions/V1Topic'
+      responses:
+        '200':
+          description: The topic was successfully updated
+        '404':
```
Comment (malinink) [inline src/main/resources/api/api.yaml]: oreder responses codes, 403 also possible

File: src/main/resources/api/api.yaml [hunk 8]
```diff
@@ -186,6 +186,132 @@ paths:
...
+          schema:
+            $ref: '#/definitions/V1Error'
+
+  /v1/topics/{topicId}:
+    parameters:
+      - name: topicId
+        in: path
+        required: true
+        description: An id of the topic
+        type: string
```
Comment (malinink) [inline src/main/resources/api/api.yaml]: integer!

File: src/main/resources/api/api.yaml [hunk 9]
```diff
@@ -186,6 +186,132 @@ paths:
...
+      summary: Get a topic with the specific id
+      operationId: getTopicById
+      tags:
+        - topics
+      responses:
+        '200':
+          description: Expected response to a valid request. Return the topic
+          schema:
+            $ref: '#/definitions/V1Topic'
+        '403':
```
Comment (malinink) [inline src/main/resources/api/api.yaml]: 400 also

File: src/main/resources/api/api.yaml [hunk 10]
```diff
@@ -186,6 +186,132 @@ paths:
...
+            $ref: '#/definitions/V1Error'
+    'delete':
+      summary: Delete a topic with the specific id
+      operationId: deleteTopicById
+      tags:
+        - topics
+      responses:
+        '200':
+          description: Successfully deleted
+        '403':
```
Comment (malinink) [inline src/main/resources/api/api.yaml]: 400 also

File: src/main/resources/api/api.yaml [hunk 11]
```diff
@@ -494,6 +620,25 @@ definitions:
   #
   # Topics
   #
+  V1Topic:
+    type: object
+    required:
+      - id
```
Comment (malinink) [inline src/main/resources/api/api.yaml]: not required

File: src/main/resources/api/api.yaml [hunk 12]
```diff
@@ -494,6 +620,25 @@ definitions:
...
+      - name
+      - subjectID
+    properties:
+      id:
+        type: integer
+        format: int64
+      name:
+        type: string
+      subjectID:
```
Comment (malinink) [inline src/main/resources/api/api.yaml]: just subject, it must be V1Identity object

File: src/main/resources/api/api.yaml [hunk 13]
```diff
@@ -494,6 +620,25 @@ definitions:
...
+        $ref: '#/definitions/V1Identity'
+    example:
+      id: 322
+      name: Database normalization
+      subjectID: 1
```
Comment (malinink) [inline src/main/resources/api/api.yaml]: fix that one too

**Review by malinink — CHANGES_REQUESTED** [review]

File: src/main/resources/api/api.yaml (hunk 3 above)
Comment (Ielay) [reply src/main/resources/api/api.yaml]: @malinink could you remind me a case that use query parameter?

Comment (Ielay) [reply src/main/resources/api/api.yaml]: @malinink I recalled it

**Comment by Ielay** [comment]
@malinink 
Everything seems to be done

File: src/main/resources/api/api.yaml [hunk 14]
```diff
@@ -494,6 +651,28 @@ definitions:
   #
   # Topics
   #
+  V1Topic:
+    type: object
+    required:
+      - name
+      - V1Identity
```
Comment (malinink) [inline src/main/resources/api/api.yaml]: subject

File: src/main/resources/api/api.yaml [hunk 15]
```diff
@@ -494,6 +651,28 @@ definitions:
...
+    properties:
+      id:
+        type: integer
+        format: int64
+      name:
+        type: string
+      V1Identity:
```
Comment (malinink) [inline src/main/resources/api/api.yaml]: subject

File: src/main/resources/api/api.yaml [hunk 16]
```diff
@@ -494,6 +651,28 @@ definitions:
...
+        $ref: '#/definitions/V1Identity'
+    example:
+      id: 322
+      name: Database normalization
+      V1Identity:
```
Comment (malinink) [inline src/main/resources/api/api.yaml]: subject

**Review by malinink — CHANGES_REQUESTED** [review]

**Comment by Ielay** [comment]
@malinink просто поменял V1Identity на subject там, где Вы отметили, и теперь одна из проверок падает. Копирую в swagger editor, так всё работает замечательно и не ругается. Не знаю, чем coveralls недоволен.

**Comment by malinink** [comment]
@Ielay please apply changes from last commit 02e9cb5032954770e2c2312788204016b31fdad0

**Comment by malinink** [comment]
And don't worry about CC.

**Pull request closed without merging**